Classes/common/HTTP/CDTURLSessionTask.m

```diff
@@ -67,24 +79,7 @@ - (void)resume
 {
     NSURLSessionTask *t = self.inProgressTask;
     if (!t) {
-        CDTHTTPInterceptorContext *ctx =
-            [[CDTHTTPInterceptorContext alloc] initWithRequest:[self.request mutableCopy]];
-
-        // We make sure all objects support `interceptRequestInContext:` during init.
-        for (NSObject<CDTHTTPInterceptor> *obj in self.requestInterceptors) {
-            ctx = [obj interceptRequestInContext:ctx];
-        }
-
-        __weak CDTURLSessionTask *weakSelf = self;
-        self.inProgressTask = [self.session
-            dataTaskWithRequest:ctx.request
-              completionHandler:^void(NSData *data, NSURLResponse *response, NSError *error) {
-                  __strong CDTURLSessionTask *self = weakSelf;
-                  if (self && self.completionHandler) {
-                      data = [NSData dataWithData:data];
-                      self.completionHandler(data, response, error);
-                  }
-              }];
+        [self makeRequest];
```

Comment:
It's not clear that `makeRequest` assigns self.inProgressTask -- instead return the task from `makeRequest` and assign in this method rather than hiding that logic in several places.